ui/src/features/Editor/components/OverlayUI/components/DebugPanel/index.tsx

```diff
@@ -53,6 +53,30 @@ const DebugPanel: React.FC = () => {
   } = useHooks();
   const t = useT();
   const [tabValue, setTabValue] = useState("debug-logs");
+
+  const [selectedFeature, setSelectedFeature] = useState<any>(null);
+  const [enableClustering, setEnableClustering] = useState<boolean>(true);
+  const [shouldFlyToFeature, setShouldFlyToFeature] = useState<boolean>(false);
```

Comment:
The way you are using this `shouldFlyToFeature` is a very precarious approach and should be avoided if possible.
Since maplibre accepts a Ref, you can define it high enough that you can use its methods inside of callbacks. Then, pass the ref down into Maplibre, and from the ref use fly to inside a callback that'll be run on the double click. I know this is a little tricky so if you don't quite understand I can try to explain on zoom and/or we can pair program it.